examples/NonReg/SPI_loop/SPI_loop.ino

```diff
@@ -0,0 +1,81 @@
+/*
+  SPI_loopback
+
+  Test the loopback transfer on SPI serial link.
+  A test string is sent 9 times and compared to the Rx frame.
+  when the MISO pin is physically connected to the MOSI pin of the SPI bus
+  Default test the SPI_A link. To test the SPI_B transfer, just define SPI_B
+  to change the pin
+
+  This example code is in the public domain.
+
+*/
+#include "SPI.h"
+
+#if defined(SPI_B)
+/* when testing the SPI _B */
+ int MOSI_PIN = 22; // to be connected to the B_MISO
+ int SCK_PIN = 23;
+ int CS_PIN = 24;
+ int MISO_PIN = 25; // to be connected to the B_MOSI
+#else
+#define MOSI_PIN  PIN_SPI_MOSI
+#define SCK_PIN   PIN_SPI_SCK
+#define CS_PIN    PIN_SPI_SS
+#define MISO_PIN  PIN_SPI_MOSI
+#endif /* SPI_B */
+
+#define BUF_SIZE 36
+uint8_t buffer_tx[] = "Thequickbrownfoxjumpsoverthelazydog\0";
+uint8_t buffer_rx[BUF_SIZE] = {};
+
+volatile uint8_t received; // SPI transfer loop counter
+volatile bool tested; // test completed
+
+void setup() {
+  received = 0; // NB of loop for spi transfer
+  tested = false;
+
+  /* configuring SPI pins*/
+  pinMode(MISO_PIN, OUTPUT);
+  pinMode(SCK_PIN, OUTPUT);
+  pinMode(CS_PIN, OUTPUT);
+  pinMode(MOSI_PIN, INPUT);
+
+#if defined(SPI_B)
+  SPI.setMISO(MISO_PIN);
+  SPI.setMOSI(MOSI_PIN);
+  SPI.setSCLK(SCK_PIN);
+  SPI.setSSEL(CS_PIN);
+
+#endif /* SPI_B */
+   SPI.begin(CS_PIN); // waking up SPI bus
+  // SPI.setClockDivider(CS_PIN, 256);     // setting a slower clock
+
+  pinMode(LED_BUILTIN, OUTPUT);   // prepare led for status
+  digitalWrite(LED_BUILTIN, LOW); // start with led off
+ }                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   
+
+void loop() {
+  if (received < 9 && !tested) { // data to transfer, 9 times
+    SPI.transfer(CS_PIN, buffer_tx, buffer_rx, BUF_SIZE, SPI_LAST);           
+    delay(100);
+    /* compare what is received to what was sent */
+    if (memcmp(buffer_tx, buffer_rx, BUF_SIZE)) {
+      /* rx buffer is not correct */
+       tested = true;
+    } else {
+      /* this transfer passed */
+      received++;
+      memset(buffer_rx, 0, BUF_SIZE);
+    }
+  }
+
+  if ((received < 9) && tested) {
```

Comment:
The status is not cleared as it loop again.
I guess a simple for loop to manage  all the transfer and finally provides the status.